torch/csrc/jit/passes/onnx/scalar_type_analysis.cpp

```diff
@@ -110,10 +110,10 @@ static c10::optional<c10::ScalarType> InferExpectedScalarType(const Node* n) {
         } else if (nkind == onnx::Constant) {
           typesFromScalars.emplace_back(
               input->node()->t(attr::value).scalar_type());
-        } else if (
-            auto scalar_type =
-                input->type()->cast<TensorType>()->scalarType()) {
-          typesFromTensors.emplace_back(*scalar_type);
+        } else if (auto tensor_type = input->type()->cast<TensorType>()) {
+          auto scalar_type = tensor_type->scalarType();
+          if (scalar_type)
+            typesFromTensors.emplace_back(*scalar_type);
         }
```

Comment:
Similar as below, do we need to consider `ListType`?

Reply:
I'm not sure, my understanding is that ListType should even be here, by that I mean shape inference should already 'converted' it to tensor. We are only concerned about the case when shape inference fails